File: src/my_py_pkg/my_py_pkg/simple_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleNode(Node):
    def __init__(self):
        super().__init__('simple_node')
        self.publisher_ = self.create_publisher(String, 'chatter', 10)
        self.timer = self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = 'Hello, ROS 2!'
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publishing: "{msg.data}"')

File: src/my_cpp_pkg/src/simple_node.cpp
#include "my_cpp_pkg/simple_node.hpp"

SimpleNode::SimpleNode() : Node("simple_node")
{
  //int unused_var = 42;  // 未使用 → warning
  publisher_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);
  timer_ = this->create_wall_timer(
    std::chrono::seconds(1),
    [this]() {
      auto msg = std_msgs::msg::String();
      msg.data = "Hello, ROS 2!";
      publisher_->publish(msg);
      RCLCPP_INFO(this->get_logger(), "Published: '%s'", msg.data.c_str());
    });
}
